btp-document/Smart-Contracts-ICON.md

```diff
@@ -0,0 +1,387 @@
+## Smart Contracts on ICON Deployment
+
+____
+
+<p align="center">
+  <img src="./images/Deploy-Contracts-ICON.png" width="800" height="300" />
+</p>
+
+Before deploying these contracts, please make sure you have already started a local node in the ICON Network. If you have not yet done this step, please check out this [link](BTP-Development-Instructions.md#1-deploy-icon-node) and accomplish it
+
+Clone this BTP project and copy this [[Java-Folder](./files/javascore.tar)] to your local folder that has created when you started your local node
+
+```shell
+$ cd INode
+
+$ git clone https://github.com/icon-project/btp.git
+
+$ cd btp && git checkout icondao
+
+# Copy javascore.tar to $CONFIG_DIR and extract
+$ tar -xzvf javascore.tar
+```
+
+### 1. Preparation - Compiling Requiring Java Files
+
+____
+
+- Run below commands to compile Java files and generate requiring `.jar` files:
+
+```shell
+$ CONFIG_DIR=/path/to/config/folder/INode
+
+$ cd $CONFIG_DIR/btp/javascore/bmv/eventDecoder
+
+$ gradle buildKusamaDecoder
+
+$ gradle buildMoonriverDecoder
+
+$ cd $CONFIG_DIR/btp/javascore/bmv/parachain
+
+$ gradle optimizedJar
+#  The commands above compiles all java files to build BMV contract
+
+#  Next, move on compiling files to build BMC contract
+$ cd $CONFIG_DIR/javascore/bmc
```

Comment:
Should be `$CONFIG_DIR/btp/javascore/bmc` ?

Reply:
that folder is extracted from `javascore.tar`.

Reply:
The final documentation will update the link to retrieve those

Reply:
Right, but when we extract the .tar, we are in btp folder.

Reply:
Sorry for confusion instructions. As I provided, `javascore.tar` should be copied to `$CONFIG_DIR` folder, then extract it. As a result, this `javascore` folder, which contains `bmc`, `nativecoinbsh`, and `irc31token`, locates at $CONFIG_DIR directory (`$CONFIG_DIR/javascore`). On the other hand, the second `javascore` folder, which was retrieved via `git clone`, has a different path - `$CONFIG_DIR/btp/javascore`

Reply:
I'm gonna make instructions at this point clearer to avoid mistake and confusion